core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStore.java

```diff
@@ -297,14 +336,60 @@ private void createSchemaTopic() throws StoreInitializationException {
                + desiredReplicationFactor + ". If this is a production environment, it's "
                + "crucial to add more brokers and increase the replication factor of the topic.");
     }
-    Properties schemaTopicProps = new Properties();
-    schemaTopicProps.put(LogConfig.CleanupPolicyProp(), "compact");
 
+    NewTopic schemaTopicRequest = new NewTopic(topic, 1, (short) schemaTopicReplicationFactor);
+    schemaTopicRequest.configs(Collections.singletonMap(LogConfig.CleanupPolicyProp(), "compact"));
     try {
-      AdminUtils.createTopic(zkUtils, topic, 1, schemaTopicReplicationFactor, schemaTopicProps,
-                             RackAwareMode.Enforced$.MODULE$);
-    } catch (TopicExistsException e) {
-      // This is ok.
+      admin.createTopics(Collections.singleton(schemaTopicRequest)).all().get(30, TimeUnit.SECONDS);
+    } catch (ExecutionException e) {
+      if (e.getCause() instanceof TopicExistsException) {
+        // This is ok.
+      } else {
+        throw e;
+      }
+    }
+  }
+
+  private void verifySchemaTopic(AdminClient admin) throws StoreInitializationException,
+                                                           InterruptedException,
+                                                           ExecutionException,
+                                                           TimeoutException {
+    log.info("Validating schemas topic {}", topic);
+
+    Set<String> topics = Collections.singleton(topic);
+    Map<String, TopicDescription> topicDescription = admin.describeTopics(topics)
+        .all().get(30, TimeUnit.SECONDS);
+
+    TopicDescription description = topicDescription.get(topic);
+    if (description.partitions().size() != 1) {
+      log.error("The schema topic " + topic + " should have only 1 partition.");
```

Comment:
Yeah, I had been preserving the existing behavior, but I've change this since if you have more partitions you'll just end up with weird behavior.